atd-etl/data_model/utils/files.py

```diff
@@ -0,0 +1,163 @@
+import os
+import subprocess
+
+from boto3 import client, resource
+
+from utils.logging import get_logger
+
+from utils.settings import LOCAL_EXTRACTS_DIR
+
+ENV = os.environ["ENV"]
+BUCKET_NAME = os.environ["BUCKET_NAME"]
+EXTRACT_PASSWORD = os.environ["EXTRACT_PASSWORD"]
+
+logger = get_logger()
+s3_client = client("s3")
+s3_resource = resource("s3")
+
+
+def format_megabytes(bytes):
+    """Format an integer as megabytes"""
+    return f"{round(bytes/1000000)}mb"
+
+
+def get_extract_name_from_file_name(file_key):
+    """Parse the full extract name from the file name"""
+    return os.path.basename(file_key).replace(".zip", "")
+
+
+def get_unzipped_extracts_local():
+    """Gets a list of sudirectories in the <LOCAL_EXTRACTS_DIR>, making the assumption that any subdirectory
```

Comment:
i initially wrote this by making use of temporary directories, but i found that to be pretty complicated to implement in combination with the option of supplying a zip file locally, and i also found it quite annoying to debug the PDF processing when files were being saved to an ephemeral temp dir.

for the production Airflow + docker deployment, local directories will essentially be treated the same as temporary directories, since containers will be removed after each run. so i think this is ok?

Reply:
nice—thanks for doing that research 🙌